### PR TITLE
expose scheduler to Task

### PR DESCRIPTION
**Pull request by aniketmaurya**
## What does this PR do?
https://github.com/PyTorchLightning/lightning-flash/discussions/175

Fixes # (issue)

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? [not needed for typos/docs]
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: flash/core/model.py [hunk 1]
```diff
@@ -156,8 +160,13 @@ def predict(
         output = data_pipeline.uncollate_fn(predictions)  # TODO: pass batch and x
         return output
 
-    def configure_optimizers(self) -> torch.optim.Optimizer:
-        return self.optimizer_cls(filter(lambda p: p.requires_grad, self.parameters()), lr=self.learning_rate)
+    def configure_optimizers(
+            self
+    ) -> Union[Tuple[Tuple[torch.optim.Optimizer], SchedulerType], Tuple[torch.optim.Optimizer]]:
+        optimizers = self.optimizer_cls(filter(lambda p: p.requires_grad, self.parameters()), lr=self.learning_rate),
+        if self.schedulers:
```
Comment (tchaton) [inline flash/core/model.py:167]: What happens if the scheduler needs to wrap the optimizer(s) ? Any solution for it ?

Comment (aniketmaurya) [reply flash/core/model.py:167]: TBH I am not sure about this. In LightningModule either we return "list of optimizer and scheduler" or only optimizer. So, my thinking was that we can do the same here. @tchaton

Comment (aniketmaurya) [reply flash/core/model.py:167]: it would be helpful for me if you can provide some reference for scheduler wrapping the optimizer.

File: flash/core/model.py [hunk 2]
```diff
@@ -55,18 +57,20 @@ class Task(pl.LightningModule):
...
-        loss_fn: Optional[Union[Callable, Mapping, Sequence]] = None,
-        optimizer: Type[torch.optim.Optimizer] = torch.optim.Adam,
-        metrics: Union[pl.metrics.Metric, Mapping, Sequence, None] = None,
-        learning_rate: float = 5e-5,
+            self,
+            model: Optional[nn.Module] = None,
+            loss_fn: Optional[Union[Callable, Mapping, Sequence]] = None,
+            optimizer: Type[torch.optim.Optimizer] = torch.optim.Adam,
+            schedulers: Optional[Any] = None,
+            metrics: Union[pl.metrics.Metric, Mapping, Sequence, None] = None,
```
Comment (tchaton) [inline flash/core/model.py:65]:
```suggestion
            optimizer: Type[torch.optim.Optimizer] = torch.optim.Adam,
            scheduler:  Optional[Type[torch.optim.lr_scheduler._LRScheduler]] = None,
            metrics: Union[pl.metrics.Metric, Mapping, Sequence, None] = None,
```

File: flash/core/model.py [hunk 3]
```diff
@@ -55,18 +57,20 @@ class Task(pl.LightningModule):
...
+            learning_rate: float = 5e-5,
     ):
         super().__init__()
         if model is not None:
             self.model = model
         self.loss_fn = {} if loss_fn is None else get_callable_dict(loss_fn)
         self.optimizer_cls = optimizer
+        self.schedulers = schedulers
```
Comment (tchaton) [inline flash/core/model.py:73]:
```suggestion
        self.scheduler = scheduler
```

File: flash/core/model.py (hunk 1 above)
Comment (tchaton) [reply flash/core/model.py:167]:
```suggestion
        if self.scheduler:
```

File: flash/core/model.py [hunk 4]
```diff
@@ -156,8 +160,13 @@ def predict(
...
         return output
 
-    def configure_optimizers(self) -> torch.optim.Optimizer:
-        return self.optimizer_cls(filter(lambda p: p.requires_grad, self.parameters()), lr=self.learning_rate)
+    def configure_optimizers(
+            self
+    ) -> Union[Tuple[Tuple[torch.optim.Optimizer], Any], Tuple[torch.optim.Optimizer]]:
+        optimizers = self.optimizer_cls(filter(lambda p: p.requires_grad, self.parameters()), lr=self.learning_rate),
+        if self.schedulers:
+            return optimizers, self.schedulers
```
Comment (tchaton) [inline flash/core/model.py:168]:
```suggestion
            return optimizers, self.scheduler(optimizer=optimizer)
```

Comment (aniketmaurya) [reply flash/core/model.py]: this will only work if there is only one optimizer @tchaton ?

**Comment by aribornstein** [comment]
@ethanwharris @tchaton @aniketmaurya is there any progress here?

**Comment by tchaton** [comment]
Dear @aniketmaurya,

Awesome work with this PR. I took some inspirations from it and finish it in this PR: https://github.com/PyTorchLightning/lightning-flash/pull/232.

Closing this one.
Best,
T.C
